src/validations/passport-gated/index.ts

```
@@ -27,46 +50,158 @@ function hasValidIssuanceAndExpiration(credential, proposalTs) {
+    scoreData.detail !== PASSPORT_NOT_SUBMITTED_ERROR &&
+    // Workaround: sometimes the API returns a 400 (Bad Request) error
+    // because the passport has never been evaluated by the Scorer before
+    scoreResponse.status !== 400
+  ) {
+    const reason =
+      (!SCORER_ID ? 'SCORER_ID missing' : scoreData.detail) ||
+      scoreResponse.statusText;
+    console.log('[passport] Scorer error', scoreData || reason);
+    throw new Error(`Scorer error: ${reason}`);
+  }
+
+  // If first time using scorer, address needs to submit passport for scoring
+  if (
+    (!scoreResponse.ok && scoreData.detail === PASSPORT_NOT_SUBMITTED_ERROR) ||
+    // Workaround: same as specified above
+    scoreResponse.status === 400
+  ) {
+    const submittedPassport = await fetch(POST_SUBMIT_PASSPORT_URI, {
+      headers,
+      method: 'POST',
+      body: JSON.stringify({ address: currentAddress, scorer_id: SCORER_ID })
+    });
+    const submissionData = await submittedPassport.json();
+
+    // Scorer done calculating passport score during submission
+    if (submittedPassport.ok && submissionData.status === 'DONE') {
+      return evalPassportScore(submissionData, scoreThreshold);
+    }
+  }
+
+  // Passport Score was already calculated
+  if (scoreResponse.ok && scoreData.status === 'DONE') {
+    return evalPassportScore(scoreData, scoreThreshold);
+  }
+
+  // Try to fetch Passport Score if still processing (scoreData.status === 'PROCESSING')
+  for (let i = 0; i < PASSPORT_SCORER_MAX_ATTEMPTS; i++) {
+    const scoreResponse = await fetch(GET_PASSPORT_SCORE_URI + currentAddress, {
+      headers
+    });
+    const scoreData = await scoreResponse.json();
+
+    if (scoreResponse.ok && scoreData.status === 'DONE') {
+      return evalPassportScore(scoreData, scoreThreshold);
+    }
+    console.log(
+      `[passport] Waiting for scorer... (${i}/${PASSPORT_SCORER_MAX_ATTEMPTS})`
+    );
+    await snapshot.utils.sleep(3e3);
+  }
+  return false;
```

If we still wait for scorer after 2 attempts, we should return an error instead of `false`